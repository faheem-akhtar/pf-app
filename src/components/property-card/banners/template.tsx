import { FunctionComponent } from 'react';

import { domClassMerge } from 'helpers/dom/class-merge';

import styles from '../property-card.module.scss';
import { PropertyCardBannersTemplatePropsInterface } from './template-props.interface';

export const PropertyCardBannersTemplate: FunctionComponent<PropertyCardBannersTemplatePropsInterface> = ({
  banners = [],
  showBanners = true,
}): JSX.Element | null => {
  if (!banners.length) {
    return null;
  }

  return (
    <ul
      className={domClassMerge(styles['banner-list'], {
        [styles['banner-list--hidden']]: !showBanners,
      })}
    >
      {banners.map(({ text, colorClass }, index) => (
        <li key={index} className={domClassMerge(styles.banner, styles[colorClass])}>
          {text}
        </li>
      ))}
    </ul>
  );
};
