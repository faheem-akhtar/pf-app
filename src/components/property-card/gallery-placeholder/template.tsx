import { FunctionComponent } from 'react';

import styles from './template.module.scss';

export const PropertyCardGalleryPlaceholderTemplate: FunctionComponent = () => (
  <div className={styles.container} data-testid='gallery-placeholder-image'>
    <svg className={styles.svg} viewBox='73 47 113 90'>
      <g>
        <path
          d='M186.39489,56.7832365 L186.39489,128.216763 C186.39489,130.801529 185.474537,133.014265 183.63389,134.854912 C181.793243,136.695559 179.580507,137.615912 176.995741,137.615912 L83.0042586,137.615912 C80.4194928,137.615912 78.2067571,136.695559 76.3661101,134.854912 C74.5254632,133.014265 73.6051103,130.801529 73.6051103,128.216763 L73.6051103,56.7832365 C73.6051103,54.1984708 74.5254632,51.985735 76.3661101,50.1450881 C78.2067571,48.3044411 80.4194928,47.3840882 83.0042586,47.3840882 L176.995741,47.3840882 C179.580507,47.3840882 181.793243,48.3044411 183.63389,50.1450881 C185.474537,51.985735 186.39489,54.1984708 186.39489,56.7832365 Z M176.995741,54.9034069 L83.0042586,54.9034069 C82.4951185,54.9034069 82.0545334,55.0894513 81.6825034,55.4614813 C81.3104733,55.8335113 81.1244289,56.2740964 81.1244289,56.7832365 L81.1244289,128.216763 C81.1244289,128.725904 81.3104733,129.166489 81.6825034,129.538519 C82.0545334,129.910549 82.4951185,130.096593 83.0042586,130.096593 L176.995741,130.096593 C177.504882,130.096593 177.945467,129.910549 178.317497,129.538519 C178.689527,129.166489 178.875571,128.725904 178.875571,128.216763 L178.875571,56.7832365 C178.875571,56.2740964 178.689527,55.8335113 178.317497,55.4614813 C177.945467,55.0894513 177.504882,54.9034069 176.995741,54.9034069 Z M171.356252,96.2596593 L171.356252,122.577275 L88.6437476,122.577275 L88.6437476,111.298297 L107.442044,92.5 L116.841192,101.899148 L146.918467,71.8218738 L171.356252,96.2596593 Z M111.201703,73.7017034 C111.201703,76.8347333 110.105117,79.4978645 107.912002,81.6909795 C105.718887,83.8840945 103.055755,84.9806814 99.9227255,84.9806814 C96.7896957,84.9806814 94.1265645,83.8840945 91.9334495,81.6909795 C89.7403344,79.4978645 88.6437476,76.8347333 88.6437476,73.7017034 C88.6437476,70.5686736 89.7403344,67.9055424 91.9334495,65.7124274 C94.1265645,63.5193124 96.7896957,62.4227255 99.9227255,62.4227255 C103.055755,62.4227255 105.718887,63.5193124 107.912002,65.7124274 C110.105117,67.9055424 111.201703,70.5686736 111.201703,73.7017034 Z'
          fill='currentColor'
        />
      </g>
    </svg>
  </div>
);
