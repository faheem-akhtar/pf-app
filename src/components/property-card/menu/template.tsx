import { FunctionComponent } from 'react';

import {
  IconSolidFavoriteTemplate,
  IconThickFavoriteTemplate,
  IconThinFavoriteTemplate,
  IconThinMenuTemplate,
} from 'components/icon';
import { domClassMerge } from 'helpers/dom/class-merge';

import styles from '../property-card.module.scss';
import { propertyCardVariantsModernIsActive } from '../variants/modern/is-active';
import { PropertyCardMenuTemplatePropsInterface } from './template-props.interface';

export const PropertyCardMenuTemplate: FunctionComponent<PropertyCardMenuTemplatePropsInterface> = ({
  cardType,
  saved,
  onSaveButtonClick,
  onMenuButtonClick,
}) => {
  const FavoriteIcon = saved
    ? IconSolidFavoriteTemplate
    : propertyCardVariantsModernIsActive(cardType)
    ? IconThickFavoriteTemplate
    : IconThinFavoriteTemplate;

  return (
    <div className={styles.menu}>
      {/*  TODO[CX-946] Extend library button types support and implement here */}
      <button
        className={domClassMerge(styles.button, styles['button--saved'])}
        onClick={onSaveButtonClick}
        data-testid='property-save-button'
      >
        <FavoriteIcon class={domClassMerge(styles['icon--saved'], { [styles['icon--saved-filled']]: saved })} />
      </button>

      <button className={domClassMerge(styles.button, styles['button--menu'])} onClick={onMenuButtonClick}>
        <IconThinMenuTemplate clipped={false} />
      </button>
    </div>
  );
};
