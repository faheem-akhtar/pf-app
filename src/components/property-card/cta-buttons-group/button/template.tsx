import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonIconPositionEnum } from 'library/button/icon-position.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { PropertyCardCtaButtonsGroupButtonTemplatePropsInterface } from './template-props.interface';

import styles from '../property-cta-buttons.module.scss';

export const PropertyCardCtaButtonsGroupButtonTemplate: React.FunctionComponent<PropertyCardCtaButtonsGroupButtonTemplatePropsInterface> =
  (props) => {
    return (
      <ButtonTemplate
        className={styles.button}
        size={ButtonSizeEnum.small}
        href={props.href}
        componentType={ButtonComponentTypeEnum.tertiary}
        icon={{ component: props.iconComponent, position: ButtonIconPositionEnum.left }}
        onClick={(e): void => {
          e.preventDefault();
          props.onClick();
        }}
      >
        {props.label}
      </ButtonTemplate>
    );
  };
