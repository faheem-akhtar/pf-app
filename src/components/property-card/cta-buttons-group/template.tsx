import { IconEmailTemplate } from 'components/icon/email-template';
import { IconSolidPhoneTemplate } from 'components/icon/solid-phone-template';
import { IconWhatsappTemplate } from 'components/icon/whatsapp-template';
import { PropertyCardCtaButtonsGroupButtonTemplate } from './button/template';
import { PropertyCardCtaButtonsGroupTemplatePropsInterface } from './template-props.interface';

import styles from './property-cta-buttons.module.scss';

export const PropertyCardCtaButtonsGroupTemplate: React.FunctionComponent<PropertyCardCtaButtonsGroupTemplatePropsInterface> =
  (props) => {
    const { contactOptions } = props;
    const { phone, whatsapp, email } = contactOptions;

    return (
      <div className={styles.container}>
        {phone && (
          <PropertyCardCtaButtonsGroupButtonTemplate
            href={phone.link}
            iconComponent={IconSolidPhoneTemplate}
            onClick={props.onCallClick}
            labelKey={'Call'}
          />
        )}
        {whatsapp && (
          <PropertyCardCtaButtonsGroupButtonTemplate
            href={whatsapp.link}
            iconComponent={IconWhatsappTemplate}
            onClick={props.onWhatsappClick}
            labelKey={'WhatsApp'}
          />
        )}
        {email && (
          <PropertyCardCtaButtonsGroupButtonTemplate
            href='#'
            iconComponent={IconEmailTemplate}
            onClick={props.onEmailClick}
            labelKey={'Email'}
          />
        )}
      </div>
    );
  };
