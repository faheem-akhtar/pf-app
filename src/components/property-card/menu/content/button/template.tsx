import { PropertyCardMenuContentButtonTemplatePropsInterface } from './template-props.interface';

import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonIconPositionEnum } from 'library/button/icon-position.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';

import styles from '../property-card-menu-content.module.scss';

export const PropertyCardMenuContentButtonTemplate = ({
  icon,
  label,
  onClick,
}: PropertyCardMenuContentButtonTemplatePropsInterface): JSX.Element => (
  <ButtonTemplate
    type='button'
    className={styles.button}
    onClick={onClick}
    componentType={ButtonComponentTypeEnum.tertiary}
    size={ButtonSizeEnum.regular}
    icon={{ component: icon, position: ButtonIconPositionEnum.left }}
  >
    {label}
  </ButtonTemplate>
);
