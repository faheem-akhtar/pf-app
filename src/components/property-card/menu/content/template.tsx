import { Fragment } from 'react';

import { IconThinReportTemplate } from 'components/icon/thin/report-template';
import { IconThinShareTemplate } from 'components/icon/thin/share-template';

import { PropertyCardMenuContentButtonTemplate } from './button/template';
import { PropertyCardMenuContentTemplatePropsInterface } from './template-props.interface';

export const PropertyCardMenuContentTemplate = ({
  socialShareOpenRef,
  reportOpenRef,
  t,
}: PropertyCardMenuContentTemplatePropsInterface): JSX.Element => {
  return (
    <Fragment>
      <PropertyCardMenuContentButtonTemplate
        label={t('cta-share')}
        icon={IconThinShareTemplate}
        onClick={(): void => {
          socialShareOpenRef.current();
        }}
      />

      {/* TODO[CX-375] - Implement hide property modal
      <PropertyCardMenuContentButtonTemplate
        label={t('cta-hide')}
        icon={IconThinEyeHideTemplate}
        onClick={(): void => undefined}
      /> */}

      <PropertyCardMenuContentButtonTemplate
        label={t('cta-report')}
        icon={IconThinReportTemplate}
        onClick={(): void => {
          reportOpenRef.current();
        }}
      />
    </Fragment>
  );
};
