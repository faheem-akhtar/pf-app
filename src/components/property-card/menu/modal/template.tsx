import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';

import styles from './property-card-menu-modal.module.scss';
import { PropertyCardMenuModalTemplatePropsBaseInterface } from './template-props-base.interface';

export const PropertyCardMenuModalTemplate = ({
  children,
  closeButtonLabel,
  onCloseButtonClick,
  onOverlayClick,
}: PropertyCardMenuModalTemplatePropsBaseInterface): JSX.Element => (
  <div
    className={styles.container}
    // Property card click was being triggered when clicking inside this template
    onClick={(e): void => {
      e.preventDefault();
      e.stopPropagation();
    }}
  >
    <button type='button' className={styles.overlay} onClick={onOverlayClick} />
    <div className={styles.content}>
      {children}

      <ButtonTemplate
        type='button'
        size={ButtonSizeEnum.regular}
        componentType={ButtonComponentTypeEnum.tertiary}
        onClick={onCloseButtonClick}
        className={styles.closeButton}
      >
        {closeButtonLabel}
      </ButtonTemplate>
    </div>
  </div>
);
