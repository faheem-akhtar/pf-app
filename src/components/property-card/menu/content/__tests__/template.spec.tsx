import { act, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { functionSelf } from 'helpers/function/self';

import { PropertyCardMenuContentTemplate } from '../template';
import { PropertyCardMenuContentTemplatePropsInterface } from '../template-props.interface';

describe('PropertyCardMenuContentTemplate', () => {
  let props: PropertyCardMenuContentTemplatePropsInterface;
  const socialShareOpenRef = { current: jest.fn() };
  const reportOpenRef = { current: jest.fn() };

  beforeEach(() => {
    props = {
      socialShareOpenRef,
      reportOpenRef,
      t: functionSelf,
    };
  });

  it('should render without throwing any errors', () => {
    const { container } = render(<PropertyCardMenuContentTemplate {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should call social share modal', () => {
    render(<PropertyCardMenuContentTemplate {...props} />);
    act(socialShareOpenRef.current);

    userEvent.click(screen.getByRole('button', { name: /share/i }));

    expect(socialShareOpenRef.current).toHaveBeenCalled();
  });

  it('should call report modal', () => {
    render(<PropertyCardMenuContentTemplate {...props} />);
    act(socialShareOpenRef.current);

    userEvent.click(screen.getByRole('button', { name: /report/i }));

    expect(reportOpenRef.current).toHaveBeenCalled();
  });
});
