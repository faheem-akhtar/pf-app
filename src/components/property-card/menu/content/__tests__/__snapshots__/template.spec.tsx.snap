// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCardMenuContentTemplate should render without throwing any errors 1`] = `
<div>
  <button
    class="button tertiary regular button"
  >
    <svg
      class="icon icon--left"
      height="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <title>
        Share
      </title>
      <path
        clip-rule="evenodd"
        d="M10.0135 4.5C10.0135 4.22386 9.78968 4 9.51354 4H8.25L8.04346 4.00521C5.78547 4.11932 4 6.09457 4 8.5V15.5L4.00488 15.7175C4.11189 18.0955 5.96457 20 8.25 20H15.75L15.9565 19.9948C18.2145 19.8807 20 17.9054 20 15.5V14.6072L19.9919 14.5173C19.9496 14.2841 19.7455 14.1072 19.5 14.1072C19.2239 14.1072 19 14.3311 19 14.6072V15.5L18.9952 15.6927C18.9021 17.5422 17.4779 19 15.75 19H8.25L8.07238 18.9948C6.36667 18.8956 5 17.3753 5 15.5V8.5L5.00483 8.30731C5.09786 6.45778 6.52213 5 8.25 5H9.51354L9.60341 4.99194C9.83666 4.94961 10.0135 4.74546 10.0135 4.5ZM16.4945 4.54821C16.664 4.37067 16.9329 4.34515 17.1308 4.47559L17.2014 4.53183L19.7383 6.95378C19.8942 7.03844 20 7.2036 20 7.39348L19.9999 7.40558C20.0024 7.51329 19.9704 7.62165 19.9039 7.71247L19.8418 7.78256L17.1979 10.2584C16.9964 10.4472 16.6799 10.4368 16.4912 10.2352C16.3234 10.0561 16.313 9.78617 16.4543 9.59583L16.5144 9.52852L18.2589 7.89348H15.5076C12.9522 7.89348 10.755 9.68821 10.6412 12.7917L10.6365 13.0462V14.855C10.6365 15.1311 10.4127 15.355 10.1365 15.355C9.89108 15.355 9.68693 15.1781 9.6446 14.9449L9.63654 14.855V13.0462C9.63654 9.28028 12.2425 6.99879 15.2941 6.89703L15.5076 6.89348H18.2273L16.5109 5.25512C16.3333 5.08562 16.3078 4.81672 16.4383 4.61878L16.4945 4.54821Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
    <div
      class="content"
    >
      cta-share
       
    </div>
  </button>
  <button
    class="button tertiary regular button"
  >
    <svg
      class="icon icon--left"
      height="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <title>
        Report
      </title>
      <path
        clip-rule="evenodd"
        d="M7.98311 20C7.98311 20.5523 6.86761 21 5.49156 21C4.11551 21 3 20.5523 3 20C3 19.5162 3.85598 19.1127 4.99325 19.02V5H11.0228C11.2537 3.85888 12.2591 3 13.4645 3H17.9493L21 7.5L17.9493 12H11.9198C11.6889 13.1411 10.6834 14 9.47805 14H5.98987V19.02C7.12714 19.1127 7.98311 19.5162 7.98311 20ZM17.4219 11L19.7947 7.5L17.4219 4H13.4645C12.6389 4 11.9696 4.67157 11.9696 5.5V11H17.4219ZM10.973 6H5.98987V13H9.47805C10.3037 13 10.973 12.3284 10.973 11.5V6Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
    <div
      class="content"
    >
      cta-report
       
    </div>
  </button>
</div>
`;
